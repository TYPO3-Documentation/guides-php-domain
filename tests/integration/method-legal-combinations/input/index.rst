=============================================
PHP method with allowed modifier combinations
=============================================

..  php:method:: doSomething1(string $whatever): string
    :abstract:

    Do something

..  php:method:: doSomething2(string $whatever): string
    :final:

    Do something

..  php:method:: doSomething3(string $whatever): string
    :public:

    Do something

..  php:method:: doSomething4(string $whatever): string
    :protected:

    Do something

..  php:method:: doSomething5(string $whatever): string
    :private:

    Do something

..  php:method:: doSomething6(string $whatever): string
    :static:

    Do something


..  php:method:: doSomething7(string $whatever): string
    :public:
    :static:

    Do something

..  php:method:: doSomething8(string $whatever): string
    :protected:
    :static:

    Do something

..  php:method:: doSomething9(string $whatever): string
    :private:
    :final:

    Do something

..  php:method:: doSomething10(string $whatever): string
    :protected:
    :abstract:

    Do something
