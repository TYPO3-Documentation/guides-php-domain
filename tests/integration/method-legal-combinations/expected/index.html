<!-- content start -->
    <div class="section" id="php-method-with-allowed-modifier-combinations">
            <h1>PHP method with allowed modifier combinations</h1>
            <dl class="php method">
    <dt class="sig sig-object php" id="dosomething1"><span class="sig-name modifier"><span class="pre">abstract</span></span>
<span class="sig-name descname"><span class="pre">doSomething1</span></span>
<span class="sig-paren">(</span>
<em class="sig-param"><span class="pre">string $whatever</span></em><span class="sig-paren">)</span>
<em class="sig-returns"><span class="pre">: string</span></em>
 <a class="headerlink" href="#dosomething1" title="Permalink to this definition">¶</a>
    </dt>
    <dd>

    <p>Do something</p>

        </dd>
</dl>
            <dl class="php method">
    <dt class="sig sig-object php" id="dosomething2"><span class="sig-name modifier"><span class="pre">final</span></span> <span class="sig-name modifier"><span class="pre">final</span></span>
<span class="sig-name descname"><span class="pre">doSomething2</span></span>
<span class="sig-paren">(</span>
<em class="sig-param"><span class="pre">string $whatever</span></em><span class="sig-paren">)</span>
<em class="sig-returns"><span class="pre">: string</span></em>
 <a class="headerlink" href="#dosomething2" title="Permalink to this definition">¶</a>
    </dt>
    <dd>

    <p>Do something</p>

        </dd>
</dl>
            <dl class="php method">
    <dt class="sig sig-object php" id="dosomething3"><span class="sig-name modifier"><span class="pre">public</span></span>
<span class="sig-name descname"><span class="pre">doSomething3</span></span>
<span class="sig-paren">(</span>
<em class="sig-param"><span class="pre">string $whatever</span></em><span class="sig-paren">)</span>
<em class="sig-returns"><span class="pre">: string</span></em>
 <a class="headerlink" href="#dosomething3" title="Permalink to this definition">¶</a>
    </dt>
    <dd>

    <p>Do something</p>

        </dd>
</dl>
            <dl class="php method">
    <dt class="sig sig-object php" id="dosomething4"><span class="sig-name modifier"><span class="pre">protected</span></span>
<span class="sig-name descname"><span class="pre">doSomething4</span></span>
<span class="sig-paren">(</span>
<em class="sig-param"><span class="pre">string $whatever</span></em><span class="sig-paren">)</span>
<em class="sig-returns"><span class="pre">: string</span></em>
 <a class="headerlink" href="#dosomething4" title="Permalink to this definition">¶</a>
    </dt>
    <dd>

    <p>Do something</p>

        </dd>
</dl>
            <dl class="php method">
    <dt class="sig sig-object php" id="dosomething5"><span class="sig-name modifier"><span class="pre">private</span></span>
<span class="sig-name descname"><span class="pre">doSomething5</span></span>
<span class="sig-paren">(</span>
<em class="sig-param"><span class="pre">string $whatever</span></em><span class="sig-paren">)</span>
<em class="sig-returns"><span class="pre">: string</span></em>
 <a class="headerlink" href="#dosomething5" title="Permalink to this definition">¶</a>
    </dt>
    <dd>

    <p>Do something</p>

        </dd>
</dl>
            <dl class="php method">
    <dt class="sig sig-object php" id="dosomething6"><span class="sig-name modifier"><span class="pre">static</span></span>
<span class="sig-name descname"><span class="pre">doSomething6</span></span>
<span class="sig-paren">(</span>
<em class="sig-param"><span class="pre">string $whatever</span></em><span class="sig-paren">)</span>
<em class="sig-returns"><span class="pre">: string</span></em>
 <a class="headerlink" href="#dosomething6" title="Permalink to this definition">¶</a>
    </dt>
    <dd>

    <p>Do something</p>

        </dd>
</dl>
            <dl class="php method">
    <dt class="sig sig-object php" id="dosomething7"><span class="sig-name modifier"><span class="pre">public</span></span> <span class="sig-name modifier"><span class="pre">static</span></span>
<span class="sig-name descname"><span class="pre">doSomething7</span></span>
<span class="sig-paren">(</span>
<em class="sig-param"><span class="pre">string $whatever</span></em><span class="sig-paren">)</span>
<em class="sig-returns"><span class="pre">: string</span></em>
 <a class="headerlink" href="#dosomething7" title="Permalink to this definition">¶</a>
    </dt>
    <dd>

    <p>Do something</p>

        </dd>
</dl>
            <dl class="php method">
    <dt class="sig sig-object php" id="dosomething8"><span class="sig-name modifier"><span class="pre">protected</span></span> <span class="sig-name modifier"><span class="pre">static</span></span>
<span class="sig-name descname"><span class="pre">doSomething8</span></span>
<span class="sig-paren">(</span>
<em class="sig-param"><span class="pre">string $whatever</span></em><span class="sig-paren">)</span>
<em class="sig-returns"><span class="pre">: string</span></em>
 <a class="headerlink" href="#dosomething8" title="Permalink to this definition">¶</a>
    </dt>
    <dd>

    <p>Do something</p>

        </dd>
</dl>
            <dl class="php method">
    <dt class="sig sig-object php" id="dosomething9"><span class="sig-name modifier"><span class="pre">final</span></span> <span class="sig-name modifier"><span class="pre">private</span></span> <span class="sig-name modifier"><span class="pre">final</span></span>
<span class="sig-name descname"><span class="pre">doSomething9</span></span>
<span class="sig-paren">(</span>
<em class="sig-param"><span class="pre">string $whatever</span></em><span class="sig-paren">)</span>
<em class="sig-returns"><span class="pre">: string</span></em>
 <a class="headerlink" href="#dosomething9" title="Permalink to this definition">¶</a>
    </dt>
    <dd>

    <p>Do something</p>

        </dd>
</dl>
            <dl class="php method">
    <dt class="sig sig-object php" id="dosomething10"><span class="sig-name modifier"><span class="pre">abstract</span></span> <span class="sig-name modifier"><span class="pre">protected</span></span>
<span class="sig-name descname"><span class="pre">doSomething10</span></span>
<span class="sig-paren">(</span>
<em class="sig-param"><span class="pre">string $whatever</span></em><span class="sig-paren">)</span>
<em class="sig-returns"><span class="pre">: string</span></em>
 <a class="headerlink" href="#dosomething10" title="Permalink to this definition">¶</a>
    </dt>
    <dd>

    <p>Do something</p>

        </dd>
</dl>
    </div>
<!-- content end -->
